src/app/(auth)/login/page.tsx

```diff
@@ -1,52 +1,89 @@
-'use client'
+// src/app/(auth)/login/page.tsx
+"use client";
 
-import { useState } from 'react'
-import { useRouter } from 'next/navigation'
-import Link from 'next/link'
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import Link from "next/link";
+import { useAuth } from "@/store/useAuth"; // 인증 스토어 임포트 zustand
+
+// JWT 로그인 API 함수
+const loginUser = async (credentials: { email: string; password: string }) => {
+  const response = await fetch("/auth/login", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(credentials),
+  });
+
+  if (!response.ok) {
+    const errorData = await response.json();
+    throw new Error(errorData.message || "로그인 실패");
+  }
+
+  return response.json();
+};
 
 export default function LoginPage() {
-  const router = useRouter()
+  const router = useRouter();
   const [formData, setFormData] = useState({
-    email: '',
-    password: '',
-  })
-  const [error, setError] = useState('')
+    email: "",
+    password: "",
+  });
+  const [error, setError] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  // 인증 스토어에서 login 액션 가져오기
+  const login = useAuth((state) => state.login);
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { name, value } = e.target
+    const { name, value } = e.target;
     setFormData((prev) => ({
       ...prev,
       [name]: value,
-    }))
-  }
+    }));
+  };
 
   const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    setError('')
+    e.preventDefault();
+    setError("");
+    setIsLoading(true);
 
     try {
-      // TODO: 실제 로그인 API 연동
-      console.info('로그인 시도:', formData)
+      // 실제 로그인 API 연동
+      const data = await loginUser(formData);
+
+      // Zustand 스토어에 로그인 정보 저장
+      login({ token: data.token, userId: data.userId });
```

Comment:
로그인 여부를 zustand store에 저장하게 되면 새로고침시 세션이 초기화되면서 로그인이 풀리게 될 것 같은데요
zustand 같은 메모리 영역보다는 cookie, storage 등의 영역에서 저장, 관리하는것이 어떨까요?

Reply:
추가로 메모리 영역에 저장한다고 하더라도 api 상태이기 때문에 굳이 zustand가 아니라 그냥 tanstack-query를 사용해도 충분해보입니다.

Reply:
제가  Zustand의 persist 미들웨어를 사용하고 있는데.persist 미들웨어는 상태를 자동으로 로컬 스토리지에 저장하고 페이지 새로고침 시 복원해주기 때문에, 이미 말씀하신 쿠키나 스토리지 관리가 적용된다고 하여 페이지를 새로고침해도 로그인 상태가 유지된다고 알고있습니다. 별도로 스토리지를 직접 관리할 필요 없이, Zustand와 persist 미들웨어 조합으로만 말이죠

Reply:
zustand 뿐만 아니라 많은 상태관리 라이브러리들이 persist 기능을 제공하고 있긴 합니다만 그럼 이제 두번째 내용에 대한 질문이 아직 붙어있긴 해요

zustand, zotai, redux 등은 기본적으로 클라이언트 상태관리 라이브러리입니다. 그런데 지금 이 내용에서 저장하고자 하는 값은 서버상태에요 (토큰, userId 등)

우리가 서버 상태관리 라이브러리로 tanstack-query를 사용하기로 결정했다면 서버 상태는 zustand 등 클라이언트 상태가 아니라 tanstack-query에서 사용하는것이 더 바람직한 관리방법이라고 생각이 듭니다

tanstack-query에서도 아마 persist 기능을 제공하고 있을거라고 생각해요 혹시 제공하고 있지 않더라도 자체 구현으로 스토리지를 사용할 수 있을것 같습니다. (tanstack-query 문서 접근이 안되서 예제를 못찾았네요 찾으면 공유드릴게요!)

서버 상태와 클라이언트 상태에 대해서 한번 다시 고민해보시면 좋을것같습니다.

```suggestion
// 대충 이런 느낌...?
      useQuery({
         queryKey: [LOGIN_PATH],
         queryFn: () => {
             isLogin? window.localStorage() : loginUser();
         }
      })
```

Reply:
쉽게 생각하면 클라이언트 상태라 하면 쉽게 생각나는게 form data(회원가입 폼)라고 보시면 되고, 서버 상태라 함은 api를 통해 서버로부터 받아온 응답 data라고 보시면 편할 것 같아요!

tanstack-query(react-query)의 core value중 하나가 바로 이 클라이언트 상태와 서버 상태를 분리해서! 관리하자입니다!

순기님의 말씀은 tanstack-query(react-query)을 사용하게 될 경우 이 token값도 결국 서버로부터 받아온 data이기 때문에 tanstack-query(react-query)로 관리하는게 적절하다고 말씀하시는 부분입니다 ㅎㅎ

아직은 tanstack-query(react-query)에 대해서 큰 이해도가 부족할 수도 있기때문에 한번 이 기회에 공부해보시는 것도 좋을 것 같아요 (실무에서 굉장히 높은 점유율을 갖는 라이브러리이기 떄문에 공부하시면 100% 도움된다고 장담합니다)

Reply:
아하 감사합니다 이번기회에 공부해보겠습니다!!
(사실 그런게있는줄도 몰랐어요..ㅎ )